datastore/tasks.js

```diff
@@ -82,128 +83,112 @@ function addTask (description, callback) {
-    }
-
-    var taskKey = datastore.key([
-      'Task',
-      taskId
-    ]);
-
-    transaction.get(taskKey, function (err, task) {
-      if (err) {
-        // An error occurred while getting the task
-        return transaction.rollback(function (_err) {
-          return callback(err || _err);
-        });
-      }
-
-      task.data.done = true;
-
-      transaction.save(task);
-
-      // Commit the transaction
-      transaction.commit(function (err) {
-        if (err) {
-          return callback(err);
-        }
+function markDone (taskId) {
+  const transaction = datastore.transaction();
+  const taskKey = datastore.key([
+    'Task',
+    taskId
+  ]);
 
-        // The transaction completed successfully.
-        console.log('Task %d updated successfully.', taskId);
-        return callback(null);
+  return transaction.run()
+    .then(() => transaction.get(taskKey))
+    .then((results) => {
+      const task = results[0];
+      task.done = true;
+      transaction.save({
+        key: taskKey,
+        data: task
       });
-    });
-  });
+      return transaction.commit();
+    })
+    .then(() => {
+      // The transaction completed successfully.
+      console.log(`Task ${taskId} updated successfully.`);
+    })
+    .catch(() => transaction.rollback());
 }
 // [END update_entity]
 
 // [START retrieve_entities]
-function listTasks (callback) {
-  var query = datastore.createQuery('Task')
+function listTasks () {
+  const query = datastore.createQuery('Task')
     .order('created');
 
-  datastore.runQuery(query, function (err, tasks) {
-    if (err) {
-      return callback(err);
-    }
+  return datastore.runQuery(query)
+    .then((results) => {
+      const tasks = results[0];
+
+      console.log('Tasks:');
+      tasks.forEach((task) => {
+        const taskKey = task[datastore.KEY];
+        console.log(taskKey.id, task);
+      });
 
-    console.log('Found %d task(s)!', tasks.length);
-    return callback(null, tasks);
-  });
+      return tasks;
+    });
 }
 // [END retrieve_entities]
 
 // [START delete_entity]
-function deleteTask (taskId, callback) {
-  var taskKey = datastore.key([
+function deleteTask (taskId) {
+  const taskKey = datastore.key([
     'Task',
     taskId
   ]);
 
-  datastore.delete(taskKey, function (err) {
-    if (err) {
-      return callback(err);
-    }
-
-    console.log('Task %d deleted successfully.', taskId);
-    return callback(null);
-  });
+  return datastore.delete(taskKey)
+    .then(() => {
+      console.log(`Task ${taskId} deleted successfully.`);
+    });
 }
 // [END delete_entity]
 
-var cli = require('yargs');
-var makeHandler = require('../utils').makeHandler;
+const cli = require(`yargs`);
+const noop = require(`../utils`).noop;
 
-var program = module.exports = {
-  addEntity: addTask,
-  updateEntity: markDone,
-  retrieveEntities: listTasks,
-  deleteEntity: deleteTask,
-  main: function (args) {
+const program = module.exports = {
+  addTask: addTask,
```

Comment:
Nit: can this be compressed by using ES6 and renaming variables? e.g. `markDone: